src/components/Button/Button.tsx

```diff
@@ -1,13 +1,90 @@
-import { Text, TouchableOpacity } from 'react-native'
+import React from 'react'
+import { MaterialIcons } from '@expo/vector-icons'
+import { View } from 'react-native'
+import {
+  Button as PaperButton,
+  ButtonProps as PaperButtonProps,
+} from 'react-native-paper'
+import { match } from 'ts-pattern'
 
-import { ButtonProps } from './Button.types'
+type IconProps =
+  | {
+      materialIconName?: keyof typeof MaterialIcons.glyphMap
+      icon?: undefined
+    }
+  | {
+      materialIconName?: undefined
+      icon?: React.ReactNode
+    }
+type ButtonProps = Omit<PaperButtonProps, 'icon' | 'children'> & {
+  onPress?: () => void
+  alignIcon?: 'left' | 'right'
+  loading?: boolean
+} & IconProps
+
+export default function Button({
+  icon,
+  materialIconName,
+  alignIcon = 'left',
+  loading = false,
+  disabled,
+  onPress,
+  children,
+  mode = 'contained',
+  ...rest
+}: React.PropsWithChildren<ButtonProps>) {
+  const matchValues = {
+    icon: Boolean(icon || materialIconName),
+    children: Boolean(children),
+  }
+
+  const iconGap = match(matchValues)
+    .with({ icon: true, children: false }, () => -16)
+    .with({ icon: true, children: true }, () => 8)
+    .with({ icon: false, children: true }, () => 0)
+    .otherwise(() => 0)
```

Comment:
Daria pra tirar um caso aqui
```suggestion
  const iconGap = match(matchValues)
    .with({ icon: true, children: false }, () => -16)
    .with({ icon: true, children: true }, () => 8)
    .otherwise(() => 0)
```